app/src/main/java/com/example/playlistmaker/player/ui/PlayerViewModel.kt

```diff
@@ -260,7 +177,61 @@ class PlayerViewModel(
         }
     }
 
+    fun setAudioPlayerManager(audioPlayerManager: AudioPlayerManager) {
+        this.audioPlayerManager = audioPlayerManager
+
+        audioPlayerManagerJob = viewModelScope.launch {
```

Comment:
Тут перед запуском джобы лучше отменить предыдущую на всякий случай, чтобы при нескольких вызовах подряд не запустить несколько подписок на флоу:
```suggestion
        audioPlayerManagerJob?.cancel()
        audioPlayerManagerJob = viewModelScope.launch {
```